### PR TITLE
DeepIntent Bid Adapter : add gpp and coppa compliance support

### PR DESCRIPTION
**Pull request by parthshah51999**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Other information

Looks like a different Adapter test failure. Not sure how to proceed here. Tried a dummy commit to rerun the pipeline and a different adapter test case failed. Any ideas ? @muuki88 @jsnellbaker @apukh-magnite @mkendall07 @ChrisHuie

### CODE REVIEW

File: modules/deepintentBidAdapter.js [hunk 1]
```diff
@@ -98,6 +99,20 @@ export const spec = {
...
+    if (bidderRequest?.gppConsent?.gppString) {
+      deepSetValue(openRtbBidRequest, 'regs.gpp', bidderRequest.gppConsent.gppString);
+      deepSetValue(openRtbBidRequest, 'regs.gpp_sid', bidderRequest.gppConsent.applicableSections);
+    } else if (bidderRequest?.ortb2?.regs?.gpp) {
+      deepSetValue(openRtbBidRequest, 'regs.gpp', bidderRequest.ortb2.regs.gpp);
+      deepSetValue(openRtbBidRequest, 'regs.gpp_sid', bidderRequest.ortb2.regs.gpp_sid);
+    }
+
+    // coppa compliance
+    if (config.getConfig('coppa') === true) {
```
Comment (ChrisHuie) [inline modules/deepintentBidAdapter.js:112]: Can you please get this from the request?

Comment (parthshah51999) [reply modules/deepintentBidAdapter.js:112]: @ChrisHuie coppa compliance looks to be read from config across adapters hence ensured this is read from config for consistency. Should I still go ahead and make the change ?

Comment (ChrisHuie) [reply modules/deepintentBidAdapter.js:112]: @parthshah51999 yes please. The majority of adapters are currently reading from config but Prebid is trying to encourage not using config for "coppa" and treating it more similarly to other regs in Prebid. We recently updated our docs on coppa -> https://docs.prebid.org/dev-docs/bidder-adaptor.html#prebid-standard-parameter-locations

Comment (parthshah51999) [reply modules/deepintentBidAdapter.js]: @ChrisHuie updated the PR. Could you please go through ?

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**